src/ast/nodes/shared/Node.ts

```diff
@@ -26,34 +86,18 @@ export default class Node {
 	trailingCommentEnd: number;
 	next: number;
 	module: Module;
-	parent: Node | Module;
+	parent: Node | { type?: string };
```

Comment:
It would be nice to be able to enforce that parent should be a Node or void.

Reply:
`void` would only be the case for `Program` nodes. Unfortunately, there are quite a few checks for parent types which would all need to be prefixed with a null-check if we allow parent to be void. This is certainly not nice but it seemed to me to have the lesser impact. Previously, there was no problem because `Module` was present as a non-undefined value though it was never used. If you have a better suggestion I am certainly open.

In my eyes, the best solution is to get rid of `parent` eventually and pass down all necessary information. This may even be advantageous if we start doing more transformations that change the (rendered) parent. But maybe in another PR.